kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetFileReader.java

```diff
@@ -60,9 +60,13 @@ public ParquetFileReader(FileIO fileIO) {
   public CloseableIterator<ColumnarBatch> read(
       FileStatus fileStatus, StructType schema, Optional<Predicate> predicate) {
 
-    final boolean hasRowIndexCol =
-        schema.indexOf(StructField.METADATA_ROW_INDEX_COLUMN_NAME) >= 0
-            && schema.get(StructField.METADATA_ROW_INDEX_COLUMN_NAME).isMetadataColumn();
+    if (Arrays.stream(MetadataColumn.values())
+        .filter(colType -> !colType.equals(MetadataColumn.ROW_INDEX))
+        .anyMatch(schema::contains)) {
+      throw new IllegalArgumentException(
+          "The parquet reader does not support metadata columns other than ROW_INDEX");
+    }
```

Comment:
I would slightly prefer this to iterate over the schema and check `isMetadataColumn` -> `metadataColumnType != ROW_INDEX`